plugins/temporal/rpc.go

```diff
@@ -0,0 +1,492 @@
+	var res interface{}
+	var err error
+
+	if in.Err != "" {
+		// complete by id with error
+		err = r.client.CompleteActivityByID(ctx, in.Namespace, in.WorkflowId, in.WorkflowRunId, in.ActivityId, &res, errors.New(in.Err))
+		if err != nil {
+			return err
+		}
+		(*out).Result = res
+		return nil
+	}
+
+	// complete without error
+	err = r.client.CompleteActivityByID(ctx, in.Namespace, in.WorkflowId, in.WorkflowRunId, in.ActivityId, &res, nil)
+	if err != nil {
+		return err
+	}
+	(*out).Result = res
+	return nil
+
+}
+
+type QueryWorkflowIn struct {
+	WorkflowId    string        `json:"wid"`
+	WorkflowRunId string        `json:"rid"`
+	QueryType     string        `json:"query_type"`
+	Args          []interface{} `json:"args"`
+}
+
+// QueryWorkflow queries a given workflow's last execution and returns the query result synchronously. Parameter workflowID
+// and queryType are required, other parameters are optional. The workflowID and runID (optional) identify the
+// target workflow execution that this query will be send to. If runID is not specified (empty string), server will
+// use the currently running execution of that workflowID. The queryType specifies the type of query you want to
+// run. By default, temporal supports "__stack_trace" as a standard query type, which will return string value
+// representing the call stack of the target workflow. The target workflow could also setup different query handler
+// to handle custom query types.
+// See comments at workflow.SetQueryHandler(ctx Context, queryType string, handler interface{}) for more details
+// on how to setup query handler within the target workflow.
+// - workflowID is required.
+// - runID can be default(empty string). if empty string then it will pick the running execution of that workflow ID.
+// - queryType is the type of the query.
+// - args... are the optional query parameters.
+// The errors it can return:
+//  - BadRequestError
+//  - InternalServiceError
+//  - EntityNotExistError
+//  - QueryFailError
+func (r *Rpc) QueryWorkflow(in QueryWorkflowIn, out *payload.RRPayload) error {
+	ctx := context.Background()
+	ev, err := r.client.QueryWorkflow(ctx, in.WorkflowId, in.WorkflowRunId, in.QueryType, in.Args...)
+	if err != nil {
+		return err
+	}
+
+	out = &payload.RRPayload{} // init and clear
+	err = ev.Get(out)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type DescribeWorkflowExecutionIn struct {
+	WorkflowId    string `json:"wid"`
+	WorkflowRunId string `json:"rid"`
+}
+
+func (r *Rpc) DescribeWorkflowExecution(in DescribeWorkflowExecutionIn, ) error {
+	ctx := context.Background()
+	// TODO really big response, we should figure out, what do we exactly need here
+	resp, err := r.client.DescribeWorkflowExecution(ctx, in.WorkflowId, in.WorkflowRunId)
+	if err != nil {
+		return err
+	}
+
+	_ = resp
+	return nil
+}
+
+// Close client and clean up underlying resources.
+func (r *Rpc) Close(_ EmptyStruct, out *EmptyStruct) error {
```

Comment:
remove it, you can't close client over RPC

Reply:
ok, removed